File: Flutter/混合开发/fluttet_boost_demo/FlutterBoostDemo/FlutterBoostDemo/ViewController.m
//
//  ViewController.m
//  FlutterBoostDemo
//
//  Created by ddhMacMini on 2020/8/3.
//  Copyright © 2020 ddhMacMini. All rights reserved.
//

#import "ViewController.h"
#import "DemoVCViewController.h"
#import "UIViewControllerDemo.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.view.backgroundColor = [UIColor yellowColor];
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.frame = CGRectMake(100, 200, 100, 100);
    btn.backgroundColor = [UIColor redColor];
    [btn setTitle:@"push native" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(pushNext) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
}

- (void)pushNext {
    NSLog(@"xxxxx");
//    DemoVCViewController *vc = [[DemoVCViewController alloc] init];
    UIViewControllerDemo *vc = [[UIViewControllerDemo alloc] initWithNibName:@"UIViewControllerDemo" bundle:[NSBundle mainBundle]];
    [self presentViewController:vc animated:YES completion:nil];
    
}

@end

File: Flutter/混合开发/Test/ios_demo/ios_demo/ViewController.swift
//
//  ViewController.swift
//  ios_demo
//
//  Created by ddhMacMini on 2020/7/28.
//  Copyright © 2020 ddhMacMini. All rights reserved.
//

import UIKit
import Flutter

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let btn: UIButton = UIButton(frame: CGRect(x: 100, y: 100, width: 120, height: 50))
        btn.backgroundColor = UIColor.red
        btn.setTitle("flutter", for: .normal)
        btn.addTarget(self, action: #selector(btnAction), for: .touchUpInside)
        self.view.addSubview(btn)
    }
    
    @objc func btnAction() {
        print("======")
        let engine : FlutterEngine = (UIApplication.shared.delegate as! AppDelegate).flutterEngine
        
        let flutterVC : FlutterViewController = FlutterViewController.init(engine: engine, nibName: nil, bundle: nil)
        flutterVC.modalPresentationStyle = .fullScreen
        self.present(flutterVC, animated: true) {
            
        }
    }

}

